packages/server/graphql/private/mutations/changeEmailDomain.ts

```diff
@@ -14,9 +15,23 @@ const changeEmailDomain: MutationResolvers['changeEmailDomain'] = async (
     throw new Error('New domain is the same as the old one')
   }
 
+  const [oldDomainUsers, newDomainUsers] = await Promise.all([
+    getUsersbyDomain(normalizedOldDomain),
+    getUsersbyDomain(normalizedNewDomain)
+  ])
+  const filteredUserIdsByOldDomain = oldDomainUsers
+    .filter((user) => {
+      const emailName = user.email.split('@')[0]
+      const nameExistsInNewDomain = newDomainUsers.find(
```

Comment:
> this'll fail if the name is included in the tld

As we're checking everything before the `@`, I'm not sure why the name being in the domain would cause it to fail? Still, as we want to return the `usersNotUpdated`, which I think makes sense, I've pushed an update that created a `proposedNewEmails` array.